File: app/src/main/java/com/example/ctap/fido2/PublicKeyCredentialRpEntity.java
package com.example.ctap.fido2;

import com.fasterxml.jackson.annotation.JsonProperty;

public class PublicKeyCredentialRpEntity {
    @JsonProperty("id") public String id;
    @JsonProperty("name") public String name;
    @JsonProperty(value = "icon", required = false) public String icon;
}

File: app/src/main/java/com/example/ctap/fido2/PublicKeyCredentialRpEntity.kt
package com.example.ctap.fido2

import com.fasterxml.jackson.annotation.JsonAutoDetect
import com.fasterxml.jackson.annotation.JsonProperty

@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
class PublicKeyCredentialRpEntity(
        @JsonProperty("id") var id: String,
        @JsonProperty("name") var name: String,
        @JsonProperty(value="icon", required=false) var icon: String?
)
